lib/realms/game.rb

```diff
@@ -1,50 +1,36 @@
 require "realms/yielder"
-require "realms/choice"
 require "realms/zones"
-require "realms/player"
 require "realms/turn"
-require "realms/trade_deck"
+require "realms/choice"
 
 module Realms
   class Game < Yielder
-    attr_reader :players, :active_turn, :trade_deck, :seed, :p1, :p2, :starter_deck
+    attr_reader :seed, :registry, :active_turn
 
     delegate :active_player, :passive_player,
-      to: :active_turn
-
-    delegate :scout, :viper,
-      to: :starter_deck
+      to: :active_turn, allow_nil: true
 
-    include Brainguy::Observable
-    include Brainguy::Observer
+    delegate :p1, :p2, :trade_deck,
+      to: :registry
 
     def initialize(seed: Random.new_seed)
       @seed = seed
-      @starter_deck = StarterDeck.new
-      @trade_deck = TradeDeck.new(self)
-      @players = ["p1", "p2"].shuffle(random: rng).map do |key|
-        instance_variable_set("@#{key}", Player.new(self, key))
-      end
-      @active_turn = Turn.first(trade_deck, p1, p2)
+      @registry = Zones::Registry.new(self)
+      registry.register!
     end
 
     def start
+      self.active_turn = Turn.first(trade_deck, p1, p2)
+      registry.register!
+      trade_deck.setup
       p1.draw(3)
       p2.draw(5)
```

Comment:
seems this stuff could just go into a setup phase that just does this stuff, or just at the top of execute.